internal/lifecycle/lifecycle.go

```diff
@@ -47,10 +48,22 @@ func (l Lifecycle) Execute(commandsDir string) error {
 // those options are set up, it falls back to reasonable defaults based on whether the sample is java-based
 // (has a pom.xml) that doesn't have a Dockerfile or isn't.
 func NewLifecycle(sampleDir, serviceName, gcrURL string) (Lifecycle, error) {
-	readmePath := filepath.Join(sampleDir, "README.md")
+	var readmePath string
+	// Searching for config file
+	if err := viper.ReadInConfig(); err == nil {
```

Comment:
At some point (doesn't need to be in this PR) you should probably consolidate all of the configuration and parse logic into a single place (probably in the `cmd` module). Otherwise as complexity grows, you'll be reimplementing and tracking the correct order of operations for configuration conflicts in multiple different locations in your code base.